src/git.ts

```diff
@@ -108,9 +108,25 @@ export async function getNewCommits(since?: Commit): Promise<CommitSequence> {
     if (since) {
         args.push(headBehindLastMigration ? 'HEAD..' + since.sha1 : since.sha1 + '..HEAD');
     }
-    const [stdout] = await execFile('git', args);
-    const output = stdout.toString().trim();
-    const commits = parseGitLog(output);
+    const commits = await (new Promise(async (resolve, reject) => {
+        const gitProcess = await spawn('git', args);
+        let buffer = '';
+        let parsedCommits = new CommitSequence();
+        gitProcess.stdout.on('data', data => {
+            buffer += data.toString().trim();
+            const completeCommits = buffer.substring(0, buffer.lastIndexOf('>>>>COMMIT\n'));
+            buffer = buffer.substring(buffer.lastIndexOf('>>>>COMMIT\n'));
+            const parsedLog = parseGitLog(completeCommits);
+            parsedCommits = new CommitSequence(...parsedCommits, ...parsedLog);
```

Comment:
V8 has a fixed argument limit of 65535. If a repo has more commits and they come in one data chunk then this will cause a Maximum call stack size exceeded error, because arguments need to be allocated on the call stack.

Reply:
Hmm. I could also push them 🤔

Reply:
that seems more performant